app/controllers/collection_numbers_controller.rb

```diff
@@ -14,16 +14,24 @@ class CollectionNumbersController < ApplicationController
   before_action :pass_query_params, except: :index
   before_action :store_location, except: [:index, :destroy]
 
-  def index # rubocop:disable Metrics/AbcSize
-    if params[:pattern].present? # rubocop:disable Style/GuardClause
-      collection_number_search and return
-    elsif params[:observation_id].present?
-      observation_index and return
-    elsif params[:by].present? || params[:q].present? || params[:id].present?
-      index_collection_number and return
-    else
-      list_collection_numbers and return
-    end
+  @index_subaction_param_keys = [
+    :pattern,
+    :observation_id,
+    :by,
+    :q,
+    :id
+  ].freeze
+
+  @index_subaction_dispatch_table = {
+    pattern: :collection_number_search,
+    observation_id: :observation_index,
+    by: :index_collection_number,
+    q: :index_collection_number,
+    id: :index_collection_number
+  }.freeze
+
+  def index
+    dispatch_to_index_subaction
   end
```

Comment:
@mo-nathan: Should I add `index` as a method of ApplicationController and get rid of `dispatch_to_index_subaction`?
The obvious advantages are shorter, DRYer code.
I choose 
```ruby
def index
    dispatch_to_index_subaction
```
because it lets readers of this file know what's going on without having to look at another file.

Reply:
I'd get rid of dispatch_to_index_subaction either by renaming it index in ApplicationController or by introducing a class between ApplicationController and these controllers, say IndexedController, that just defines index.  It depends on whether we expect all ApplicationControllers to have some index action or not.

Reply:
1. I will rename `ApplicationController#dispatch_to_index_subaction` to `index`, 
and, in the relevant subclasses delete `dispatch_to_index_subaction` and `index` (which is now redundant).
2. There are many controllers that lack any kind of `index` action.
3. I'd rather leave an intermediate controller to a future PR; I want to keep this PR relatively small and uncomplicated.